server/container/util/src/main/java/org/apache/james/util/scanner/SpamAssassinInvoker.java

```diff
@@ -78,96 +68,60 @@ public SpamAssassinInvoker(String spamdHost, int spamdPort) {
+                BufferedReader in = new BufferedReader(new InputStreamReader(socket.getInputStream()))) {
 
-            out = socket.getOutputStream();
-            in = new BufferedReader(new InputStreamReader(socket.getInputStream()));
             out.write("CHECK SPAMC/1.2\r\n\r\n".getBytes());
 
             // pass the message to spamd
             message.writeTo(out);
             out.flush();
             socket.shutdownOutput();
-            String s;
-            while ((s = in.readLine()) != null) {
-                if (s.startsWith("Spam:")) {
-                    StringTokenizer t = new StringTokenizer(s, " ");
-                    boolean spam;
-                    try {
-                        t.nextToken();
-                        spam = Boolean.valueOf(t.nextToken());
-                    } catch (Exception e) {
-                        // On exception return flase
-                        return false;
-                    }
-                    t.nextToken();
-                    hits = t.nextToken();
-                    t.nextToken();
-                    required = t.nextToken();
-
-                    if (spam) {
-                        // message was spam
-                        headers.put(FLAG_MAIL_ATTRIBUTE_NAME, "YES");
-                        headers.put(STATUS_MAIL_ATTRIBUTE_NAME, "Yes, hits=" + hits + " required=" + required);
-
-                        // spam detected
-                        return true;
-                    } else {
-                        // add headers
-                        headers.put(FLAG_MAIL_ATTRIBUTE_NAME, "NO");
-                        headers.put(STATUS_MAIL_ATTRIBUTE_NAME, "No, hits=" + hits + " required=" + required);
-
-                        return false;
-                    }
+            String line;
+            while ((line = in.readLine()) != null) {
+                if (isSpam(line)) {
+                    return processSpam(line);
                 }
             }
-            return false;
+            return SpamAssassinResult.empty();
         } catch (UnknownHostException e1) {
             throw new MessagingException("Error communicating with spamd. Unknown host: " + spamdHost);
         } catch (IOException | MessagingException e1) {
             throw new MessagingException("Error communicating with spamd on " + spamdHost + ":" + spamdPort + " Exception: " + e1);
-        } finally {
-            try {
-                Closeables.close(in, true);
-                Closeables.close(out, true);
-                socket.close();
-            } catch (Exception e) {
-                // Should never happen
-            }
-
         }
     }
 
-    /**
-     * Return the hits which was returned by spamd
-     * 
-     * @return hits The hits which was detected
-     */
-    public String getHits() {
-        return hits;
+    private SpamAssassinResult processSpam(String line) {
+        String[] split = Pattern.compile(" ").split(line);
```

Comment:
First, compiling a Pattern in a loop is not really a wise choice because it's slow.
Second, when you need to split on a char or a String (not a regex), guava Splitter is much recommanded.